src/text/raw_text_writer.rs

```diff
@@ -403,7 +467,9 @@ impl<'a, W: Write> Writer for RawTextWriter<W> {
     /// Writes the provided &str value as an Ion string.
     fn write_string<S: AsRef<str>>(&mut self, value: S) -> IonResult<()> {
         self.write_scalar(|output| {
-            write!(output, "\"{}\"", value.as_ref())?;
+            write!(output, "\"")?;
+            RawTextWriter::write_escaped_text_body(output, value)?;
+            write!(output, "\"")?;
```

Comment:
🗺️ The `RawTextWriter` now escapes characters that require it in the body of a `string`.